### PR TITLE
[RPA-1144] Bot-scheduler keep-alive message

### PR DESCRIPTION
**Pull request by Aeri4a**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

---

I agree to the terms of the RunBotics Contributor License Agreement.

### CODE REVIEW

File: runbotics/runbotics-desktop/src/core/websocket/listeners/process.listener.ts [hunk 1]
```diff
@@ -15,13 +15,14 @@ import { RunboticsLogger } from '#logger';
...
-import { StartProcessMessageBody } from './process.listener.types';
-import { MessageQueueService } from '../queue/message-queue.service';
+import { StartProcessMessageBody, KeepAliveStatus } from './process.listener.types';
+import { MessageQueueService, Message } from '../queue/message-queue.service';
 import { WebsocketService } from '../websocket.service';
 
 @Injectable()
 export class ProcessListener {
     private readonly logger = new RunboticsLogger(ProcessListener.name);
+    private keepAliveStatus: KeepAliveStatus = { intervalId: undefined, isActive: false };
```
Comment (preludium) [inline runbotics/runbotics-desktop/src/core/websocket/listeners/process.listener.ts]: move this object outside class and name it like `initKeepAliveStatus`

File: runbotics/runbotics-desktop/src/core/websocket/listeners/process.listener.ts [hunk 2]
```diff
@@ -31,12 +32,32 @@ export class ProcessListener {
...
+
+            this.websocketService.emitMessage(message, true);
+        }, 60000);
+
+        this.keepAliveStatus = { intervalId, isActive: true };
+    }
+
+    private async stopKeepingConnection() {
+        clearInterval(this.keepAliveStatus.intervalId);
+        this.keepAliveStatus = { ...this.keepAliveStatus, isActive: false };
```
Comment (preludium) [inline runbotics/runbotics-desktop/src/core/websocket/listeners/process.listener.ts]: assign here init object

File: runbotics/runbotics-desktop/src/core/websocket/listeners/process.listener.ts [hunk 3]
```diff
@@ -31,12 +32,32 @@ export class ProcessListener {
...
+                payload: null
+            };
+
+            this.websocketService.emitMessage(message, true);
+        }, 60000);
+
+        this.keepAliveStatus = { intervalId, isActive: true };
+    }
+
+    private async stopKeepingConnection() {
```
Comment (preludium) [inline runbotics/runbotics-desktop/src/core/websocket/listeners/process.listener.ts]: I dont like this naming `start/stop keeping connection`
I suggest `beginKeepAliveInterval` and `clearKeepAliveInterval`

Comment (preludium) [reply runbotics/runbotics-desktop/src/core/websocket/listeners/process.listener.ts]: additionally, what async about those methods?

File: runbotics/runbotics-scheduler/src/websocket/bot/bot.gateway.ts [hunk 4]
```diff
@@ -163,6 +168,25 @@ export class BotWebSocketGateway implements OnGatewayDisconnect, OnGatewayConnec
...
+
+                return HttpStatus.OK;
+            })
+            .catch(error => {
+                return HttpStatus.INTERNAL_SERVER_ERROR;
+            });
+
+        return result;
+    }
+
```
Comment (preludium) [inline runbotics/runbotics-scheduler/src/websocket/bot/bot.gateway.ts]: I thought more like creating a custom method updating the bot entity with a single query

bot doesn't care about response status it's more like `If you do something with it or not it's your business`

**Review by preludium — CHANGES_REQUESTED** [review]

**Review by damlit — APPROVED** [review]

**Review by preludium — APPROVED** [review]

**Pull request merged**